scout/adapter/mongo/case.py

```diff
@@ -288,28 +297,29 @@ def _update_case_id_query(self, query, id_list):
 
     def cases(
         self,
-        owner=None,
-        collaborator=None,
-        query=None,
-        skip_assigned=False,
-        has_causatives=False,
-        reruns=False,
-        rerun_monitor=False,
-        finished=False,
-        research_requested=False,
-        is_research=False,
-        has_rna_data=False,
-        status=None,
-        phenotype_terms=False,
-        group=None,
-        pinned=False,
-        cohort=False,
-        name_query=None,
-        yield_query=False,
-        within_days=None,
-        assignee=None,
-        verification_pending=None,
-        has_clinvar_submission=None,
+        owner: str = None,
+        collaborator: str = None,
+        query: Dict[str, Any] = None,
+        skip_assigned: bool = False,
+        has_causatives: bool = False,
+        reruns: bool = False,
+        rerun_monitor: bool = False,
+        finished: bool = False,
+        research_requested: bool = False,
+        is_research: bool = False,
+        has_rna_data: bool = False,
+        status: Any = None,
+        phenotype_terms: bool = False,
+        group: ObjectId = None,
+        pinned: bool = False,
+        cohort: bool = False,
+        name_query: str = None,
+        yield_query: bool = False,
+        within_days: int = None,
+        assignee: str = None,
+        verification_pending: bool = None,
+        has_clinvar_submission: bool = None,
+        projection: Dict[str, Any] = None,
     ):
```

Comment:
```suggestion
    ) -> List[dict]:
```

Reply:
Ah, it used to yield a dict. Lets see...

Reply:
Ah, sry, cases. Right, that can do either of two, depending on that yield query parameter. If I had a really quiet week I would refactor that, splitting it into two functions, with one calling the other if needed.

Reply:
I suppose it could be an optional[union[List[dict], Dict] but seriously..